### PR TITLE
Make acceptance slow tests faster

### PR DESCRIPTION
**Pull request by chadhietala**
This should make the slow acceptance tests faster. It does so by capturing the initial npm and bower installs in a tmp directory and then symlinks them back to the test target.

### CODE REVIEW

File: tests/acceptance/express-server-restart-slow.js [hunk 1]
```diff
@@ -22,61 +23,28 @@ describe('Acceptance: express server restart', function () {
...
-
-    return tmp.teardown('./common-tmp')
-      .then(function() {
-        conf.restore();
-      });
+    return teardownTestTargets();
   });
 
-  var appRoot;
+  // var appRoot;
```
Comment (jakecraige) [inline tests/acceptance/express-server-restart-slow.js]: extra comment

File: tests/helpers/acceptance.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+function linkDependencies(name) {
+  return tmp.setup('./tmp').then(function() {
+    return copy('./common-tmp/' + name, './tmp/' + name);
+  }).then(function() {
+    var nodeModulesPath = './tmp/' + name + '/node_modules/';
+    var bowerComponetsPath = './tmp/' + name + '/bower_components/';
+    mvRm(nodeModulesPath, '.node_modules-tmp/');
+    mvRm(bowerComponetsPath, '.bower_components-tmp/');
+    symLinkDir('./tmp/' + name, '.node_modules-tmp', 'node_modules');
+    symLinkDir('./tmp/' + name, '.bower_components-tmp', 'bower_components');
```
Comment (chadhietala) [inline tests/helpers/acceptance.js]: This can be cleaned up.

File: tests/helpers/acceptance.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    conf.restore();
+  });
+}
+
+function linkDependencies(name) {
+  return tmp.setup('./tmp').then(function() {
+    return copy('./common-tmp/' + name, './tmp/' + name);
+  }).then(function() {
+    var nodeModulesPath = './tmp/' + name + '/node_modules/';
+    var bowerComponetsPath = './tmp/' + name + '/bower_components/';
```
Comment (jakecraige) [inline tests/helpers/acceptance.js]: typo `bowerComponetsPath` to `bowerComponentsPath`, lol

**Comment by jakecraige** [comment]
this looks awesome. Maybe we can even start running the whole suite locally sometimes, lol

**Comment by chadhietala** [comment]
`runCommand` keeps exiting with 1. Will dig in later.

**Comment by stefanpenner** [comment]
@chadhietala this makes me extremely happy. I have had this lingering over my head for far to long. You are the man!

**Comment by joostdevries** [comment]
Awesome helper :-) Could you document the publicly exposed methods? I hope to whip up PR's for the other test helpers somewhere next month.

**Comment by chadhietala** [comment]
@stefanpenner This should be good to go once CI comes back. Ran it once already just squashed.

**Comment by chadhietala** [comment]
Shaved off about 2 minutes.

Before:



After:

**Comment by stefanpenner** [comment]
awesome, good work.

I wonder what the next biggest bottleneck is..

**Pull request merged**